### ISSUE TITLE
Contact backup first time crashes

### ISSUE THREAD

**Issue by imjarp** [nextcloud/android#1017]
### Actual behaviour
- Select backup contacts for the _first time_ ( press button back up now ) application crashes

### Expected behaviour
- Contact backup file uploads to server. 

If it helps the problem is that the method `expireFiles` tries to get the folder `/.Contacts-Backup/` before it´s creation in the `CreateFolderOperation`
 
### Steps to reproduce
1. Open app
2. Select back up contacts menu
3. Press the button backup now
4. Wait until the service starts

### Environment data
Android version: 7.0

Device model: Huawei P10

Stock or customized system:

Nextcloud app version:1.4.2

Nextcloud server version:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
05-17 15:32:20.336 20196-20196/com.nextcloud.client E/AndroidRuntime: FATAL EXCEPTION: main
                                                                Process: com.nextcloud.client, PID: 20196
                                                                java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.owncloud.android.datamodel.OCFile.getFileName()' on a null object reference
                                                                    at com.owncloud.android.services.ContactsBackupJob.expireFiles(ContactsBackupJob.java:176)
                                                                    at com.owncloud.android.services.ContactsBackupJob.access$200(ContactsBackupJob.java:62)
                                                                    at com.owncloud.android.services.ContactsBackupJob$OperationsServiceConnection.onServiceConnected(ContactsBackupJob.java:254)
                                                                    at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1586)
                                                                    at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1614)
                                                                    at android.os.Handler.handleCallback(Handler.java:761)
                                                                    at android.os.Handler.dispatchMessage(Handler.java:98)
                                                                    at android.os.Looper.loop(Looper.java:156)
                                                                    at android.app.ActivityThread.main(ActivityThread.java:6577)
                                                                    at java.lang.reflect.Method.invoke(Native Method)
                                                                    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:941)
                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:831)


```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
Thank you for your bug report!
This is fixed in #1019

**Comment by neteler** [comment]
Is fix already included in the current nextcloud-dev version on Fdroid?

**Comment by tobiasKaminsky** [comment]
Yes it is.

**Comment by neteler** [comment]
Unfortunately it still crashes when selecting "backup contacts" for the first time (Menu entry "Backup contacts"). I am using a newly installed Nextcloud 12 installation on Debian and the current nextcloud-dev from Fdroid (version 20170615). E.g. photo upload works ok.

**Comment by tobiasKaminsky** [comment]
I am sorry, I fixed only a similar bug, but omitted this one.
I have fixed this in the #1027 branch and it will be soon in fdroid.

**Comment by neteler** [comment]
In the latest nextcould-dev version on fdroid the "Contacts backup" menu entry is gone entirely...

**Comment by tobiasKaminsky** [comment]
@neteler it is now moved to settings as our designers said it is to prominent on the drawer and it better fits beneath "caldav integration".

**Comment by neteler** [comment]
@tobiasKaminsky ok thanks. However, in this new version nextcould-dev crashes when clicking on "Settings" so that I couldn't test again the "Contacts backup"

**Comment by tobiasKaminsky** [comment]
@neteler Strange, this is working fine for me. Which version do you use? And which android device?

**Comment by neteler** [comment]
@tobiasKaminsky I'm using version 20170623 on a Fairphone FPU1, Android 4.2.2

**Comment by tobiasKaminsky** [comment]
I fixed the crash, but it seems that it will be on fdroid with at least version 20170627 (build from tomorrow)

**Comment by neteler** [comment]
I updated to today's version, rebooted and it crashes when clicking on "Settings" (in order test again the "Contacts backup"). Sorry for no better news yet

**Comment by tobiasKaminsky** [comment]
This is strange. I am testing it with an Android 4.0.4 with 20170627.
Do you can get me a logcat? https://wiki.lineageos.org/logcat.html

**Comment by neteler** [comment]
The logcat I have put here: https://data.neteler.org/tmp/logcat.zip (44k).

I have also uninstalled the app and re-installed it: still crashing as before in "Settings".

**Comment by tobiasKaminsky** [comment]
In logcat it is said that you use 20170624.
Fdroid is always some time behind. 
You can test it with https://download.nextcloud.com/android/dev/ and use 20170627

**Comment by neteler** [comment]
Superb! With 20170627 the crash problems are gone.

**Comment by tobiasKaminsky** [comment]
Thanks for confirmation.
(we will close this once the PR is merged into master)

**Comment by tobiasKaminsky** [comment]
#1027 merged -> closing
